packages/p2p/app/index.js

```diff
@@ -0,0 +1,68 @@
+/* P2P peer discovery server */
+
+const WSServer = require('rpc-websockets').Server
+
+function conf (name, defaultValue) {
+  var value = process.env[name]
+  if (value === undefined) {
+    value = defaultValue
+  }
+  if (value === undefined) {
+    throw new Error(`Required configuration "${name}" not found.`)
+  }
+  return value
+}
+
+const host = conf('PEER_DISCOVERY_HOST', 'localhost')
+const port = conf('PEER_DISCOVERY_PORT', '8889')
+
+const server = new WSServer({ host, port })
+
+console.log(`Peer discovery server listening on ws://${host}:${port}`)
+
+const ROOMS = {}
+
+server.register('joinRoom', ([room, id]) => {
+  if (!ROOMS[room]) { ROOMS[room] = [] }
+  if (ROOMS[room][id]) {
+    console.log(`Error: ID '${id}' already in peer discovery channel '${room}'`)
+    return
+  }
+  console.log(`Id ${id} joining room ${room}`)
+  ROOMS[room].push(id)
+
+  // Time out from the room after 1 minute
+  setTimeout(() => {
+    console.log(`ID ${id} timed out from room ${room}`)
+    leaveRoom(room, id)
+  }, 240000)
```

Comment:
Let's make this configurable, for now we can make a constant at the top of the file `const ROOM_TIMEOUT=240000`